libr/core/visual.c

```diff
@@ -937,25 +942,24 @@ R_API int r_core_visual_xrefs_X(RCore *core) {
 					? "CODE (CALL)"
 					: "DATA",
 					refi->addr, fun->name, f? f->name: "");
+
+				references[count] = refi->addr;
 				if (++count > 9) {
 					break;
 				}
 			}
 		}
 	}
-	r_cons_flush ();
 	if (!count) {
-		return 0;
+		r_cons_printf (" No REF found at 0x%"PFMT64x "\n", addr);
 	}
+
+	r_cons_flush ();
 	ch = r_cons_readchar ();
-	if (fun && fun->refs) {
-		if (IS_DIGIT (ch)) {
-			refi = r_list_get_n (fun->refs, ch - 0x30);
-			if (refi) {
-				r_core_cmdf (core, "s 0x%"PFMT64x, refi->addr);
-				ret = 1;
-			}
-		}
+
+	if (IS_DIGIT (ch) && (ch - 0x30) <= count) {
+		r_core_cmdf (core, "s 0x%"PFMT64x, references[ch - 0x30]);
```

Comment:
there's no need to use cmdf for just a seek. what about using r_core_seek() instead?